02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+        |       |<-(8)--- funding_locked ------|       |
+        +-------+                              +-------+
+
+        - where node A is *opener*/*initiator* and node B is
+          *accepter*/*non-initiator*
+
+### The `open_channel2` Message
+
+This message initiates the v2 channel establishment workflow.
+
+1. type: 64 (`open_channel2`)
+2. data:
+   * [`chain_hash`:`chain_hash`]
+   * [`channel_id`:`channel_id`]
+   * [`u32`:`funding_feerate_perkw`]
+   * [`u32`:`commitment_feerate_perkw`]
+   * [`u64`:`funding_satoshis`]
+   * [`u64`:`dust_limit_satoshis`]
+   * [`u64`:`max_htlc_value_in_flight_msat`]
+   * [`u64`:`htlc_minimum_msat`]
+   * [`u16`:`to_self_delay`]
+   * [`u16`:`max_accepted_htlcs`]
+   * [`u32`:`locktime`]
+   * [`point`:`funding_pubkey`]
+   * [`point`:`revocation_basepoint`]
+   * [`point`:`payment_basepoint`]
+   * [`point`:`delayed_payment_basepoint`]
+   * [`point`:`htlc_basepoint`]
+   * [`point`:`first_per_commitment_point`]
+   * [`byte`:`channel_flags`]
+   * [`opening_tlvs`:`tlvs`]
+
+
+1. `tlv_stream`: `opening_tlvs`
+2. types:
+   1. type: 1 (`option_upfront_shutdown_script`)
+   2. data:
+       * [`u16`:`shutdown_len`]
+       * [`shutdown_len*byte`:`shutdown_scriptpubkey`]
+
+
+Rationale and Requirements are the same as for [`open_channel`](#the-open_channel-message), with the following additions:
+
+#### Requirements:
+
+If nodes have negotiated `option_dual_fund`:
+  - the opening node:
+    - MUST NOT send `open_channel`
+
+The sending node:
+  - MUST set `funding_feerate_perkw` to the feerate for this transaction
+  - MUST ensure `temporary_channel_id` is unique from any
+    other channel ID with the same peer.
+
+The receiving node:
+  - MAY fail the negotiation if:
+    - the `locktime` is unacceptable
+    - the `funding_feerate_per_kw` is unacceptable
+
+#### Rationale
+`channel_id` for the `open_channel2` MUST be derived using a zero-d out
+basepoint for the peer's revocation basepoint.  This allows the peer to
+return channel-assignable errors before the *accepter*'s revocation
+basepoint is known.
+
+`funding_feerate_perkw` indicates the fee rate that the opening node will
+pay for the funding transaction in satoshi per 1000-weight, as described
+in [BOLT-3, Appendix F](03-transactions.md#appendix-f-dual-funded-transaction-test-vectors).
+
+`locktime` is the locktime for the funding transaction.
+
+The receiving node, if the `locktime` or `feerate_funding_perkw` is considered
+out of an acceptable range, may fail the negotiation. However, it is
+recommended that the *accepter* permits the channel open to proceed
+without their participation in the channel's funding.
+
+Note that `open_channel`'s `channel_reserve_satoshi` has been omitted.
+Instead, the channel reserve is fixed at 1% of the total channel balance
+(`open_channel2`.`funding_satoshis` + `accept_channel2`.`funding_satoshis`)
+rounded down to the nearest whole satoshi or the `dust_limit_satoshis`,
+whichever is greater.
```

Comment:
Can you elaborate on why the explicit reserve was removed?
We find it useful to be able to explicitly specify it to allow setting it to `0` for mobile wallet peers (only on the wallet side), where having a channel reserve makes understanding the balance very hard for users (which means our node takes a risk that the mobile wallet peer may misbehave, but since they pay us fees it's a trade-off we choose to make for a better UX).

Reply:
We removed it so as to reduce the number of things that are communicated between peers. This assumed that every channel needed/would have a reserve requirement, and we picked a 'sane, arbitrary' default for it.

If this is a useful feature to be able to negotiate, we can certainly add it back.

Is being able to negotiate the reserve amount useful, or should we simply turn it into a flag that signals to use/not use the reserve.

Reply:
I think here I would advocate to keep the reserve negotiation. Existence of the reserve could be a data point that the game-theory backing up the LN-penalty mechanism is empirically sounds and one of the reason we see few attacks by broadcasting a revoked state today on the network.

On the otherside, I think it make sense to set up to low amounts (even `0`) where there are other out-of-band enforcement mechanism of the channel state correctness (e.g business brand of a LSP).

If a node operator is interested to benefit from the advantages of `channelv2` and would like to enforce a significant channel reserve on its peers, it could make sense to make the reserve dynamic and subject to negotiation.

Reply:
I was thinking the same @ariard, but then I realized that:

- 1% is a good default since most channels won't care
- we can introduce (later) a tlv field for optional reserve negotiation: that's what we're going to do for Phoenix

So I think it's good that the default case doesn't contain a configurable reserve, since it optimizes the most common scenario while allowing configuration via a future tlv field.

Reply:
Optional reserve negotiation SGTM!

Reply:
> 1% is a good default since most channels won't care about configuring it properly

If every wallet is going to discard the 1% reserve requirement then it should be mentioned in here that it's not a requirement. Unless you explicitly want everyone to go against the BOLT spec (which what is the purpose of a spec?), then it should be clarified.

Reply:
I disagree, wallets are a special case of a lightning node. The spec should decide what the default is, and special cases should override with extensions. It should be a wallet extension feature to use 0-reserve (which I proposed [here](https://github.com/lightning/blips/pull/30)).